ramanujan/enumerators/RelativeGCFEnumerator.py

```diff
@@ -0,0 +1,283 @@
+                continue
+
+            if key in self.hash_table:  # find hits in hash table
+                results.append(Match(key, metadata.an_coef, metadata.bn_coef))
+
+            if i == next_status_print:  # print status.
+                next_status_print += 1000
+                print(
+                    f'passed {i} out of {self.poly_domains.num_iterations} ' +
+                    f'({round(100. * i / self.poly_domains.num_iterations, 2)}%). ' +
+                    f' found so far {len(results)} results')
+                print(f'currently at an = {metadata.an_coef} bn = {metadata.bn_coef}')
+
+        if verbose:
+            print(f'created results after {time() - start}s')
+            print(f'found {len(results)} results')
+        return results
+
+    def _improve_results_precision(self, intermediate_results: List[Match], verbose=True):
+        """
+        For each results, calculate the GCD to a higher dept, and return the calculated result 
+        with the original result.
+        """
+        precise_results = []
+        counter = 0
+        n_iterations = len(intermediate_results)
+        key_factor = 10 ** g_N_verify_compare_length
+        an_series_iter, bn_series_iter = self.poly_domains.get_calculation_method()
+
+        for res in intermediate_results:
+            counter += 1
+            
+            if (counter % 10_000 == 0 or counter == n_iterations) and verbose:
+                print('Calculated {} matches out of {} to a more precise value.'.format(
+                    counter, n_iterations))
+
+            an_iter = an_series_iter(res.rhs_an_poly, SECOND_STEP_MAX_ITERS, start_n=0)
+            bn_iter = bn_series_iter(res.rhs_bn_poly, SECOND_STEP_MAX_ITERS, start_n=0)
+
+            try:
+                long_key, precision = gcf_calculation_to_precision(an_iter, bn_iter, g_N_verify_compare_length,
+                                                                   min_iters=SECOND_STEP_MIN_ITERS,
+                                                                   burst_number=SECOND_STEP_BURST_NUMBER)
+            except NotConverging as e:
+                print(f"{res} does not converge. Continuing...")
+                print(e)
+                continue
+            except (ZeroInAn, ZeroDivisionError) as e:
+                print(f" exception for {res}. Continuing...")
+                print(e)
+                continue
+
+            rhs_val = mpmath.mpf(long_key) / key_factor
+            rhs_str = mpmath.nstr(rhs_val, precision)
+
+            precise_results.append((res, rhs_str, precision))
+        return precise_results
+
+    def _refine_results(self, precise_intermediate_results, verbose=True):
+        """
+        This step is identical to the one in Efficient enumerator. The two we're quite different in the
+        rest of the implementation, so it didn't feel right to extend EfficientGCFEnumerator in this class.
```

Comment:
Efficient enumerator could use the cache method from this class, as it greatly simplifies the first enumeration.